### PR TITLE
Configure publication to Sonatype / Maven Central

### LINKED ISSUE #25: Publish to Sonatype/Maven Central
- [ ] Configure publication in CI
    - [ ] Publish to Sonatype SNAPSHOTs on merges to `main`
    - [ ] Publish to Maven Central on tag
- [ ] Add/update Gradle configuration details for `jcenter()`

### LINKED ISSUE #27: Run Apple platform tests on CI
In order to run Apple platform tests (and publish) on CI, need to configure CI to use macOS [GitHub hosted runner](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources).

### PR DESCRIPTION
**Pull request by twyatt**
Fixes #25
Fixes #27

### CODE REVIEW

**Comment by twyatt** [comment]
Note to self: consider using Gradle config provided in [examples](https://github.com/actions/cache/blob/main/examples.md#java---gradle).

Current config works, but may be too eager (possibly capture secret information in `.gradle` — we don't currently store any information in there, but it does mean we need to be careful, whereas example only pulls in safe directories).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,54 +1,53 @@
...
       - name: Gradle cache
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
-          path: ~/.gradle
-          key: grade-${{ hashFiles('**/build.gradle') }}
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.konan
```
Comment (twyatt) [inline .github/workflows/ci.yml:33]: Used by Kotlin multiplatform plugin for caching it's dependencies.

File: coroutines/README.md [hunk 2]
```diff
@@ -1,52 +1,64 @@
...
+repositories {
+    jcenter() // or mavenCentral()
+}
+
 dependencies {
     implementation("com.juul.tuulbox:coroutines-$platform:$version")
 }
 ```
 
+_Where `$platform` represents (should be replaced with) the desired platform dependency (e.g. `android`)._
```
Comment (cedrickcooke) [inline coroutines/README.md:60]: Should this actually say `jvm`? I don't think we have any android-specific compilation yet.

Comment (twyatt) [reply coroutines/README.md:60]: Good call. I was blindly copy/pasting from other projects. I'll change it to `jvm` here. 👍

**Review by cedrickcooke — APPROVED** [review]

**Review by Phoenix7351 — APPROVED** [review]

**Pull request merged**